src/api/MetadataQuery.js

```diff
@@ -147,24 +174,48 @@ class MetadataQuery extends Base {
      * @param {Object} - metadata query response object
      * @return {Object} - flattened metadata query response object
      */
-    flattenMetdataQueryResponse = ({ entries, next_marker }: MetadataQueryResponse): FlattenedMetadataQueryResponse => {
+    flattenMetdataQueryResponse = (): FlattenedMetadataQueryResponse => {
+        const { entries, next_marker }: MetadataQueryResponse = this.metadataQueryResponse;
         return {
             items: entries.map<FlattenedMetadataQueryResponseEntry>(this.flattenResponseEntry),
             nextMarker: next_marker,
         };
     };
 
     /**
-     * @param {Object} response
+     * @param {Object} template metadata template object
      */
-    queryMetadataSuccessHandler = ({ data }: { data: MetadataQueryResponse }): void => {
+    successHandler = (template: MetadataTemplate | void): void => {
+        this.templateInfo = getProp(template, 'data');
         const cache: APICache = this.getCache();
-        const filteredResponse = this.filterMetdataQueryResponse(data);
         // Flatten the filtered metadata query response and set it in cache
-        cache.set(this.key, this.flattenMetdataQueryResponse(filteredResponse));
+        const flattenedQueryResponse = this.flattenMetdataQueryResponse();
+        cache.set(this.key, flattenedQueryResponse);
         this.finish();
     };
 
+    /**
+     * Gets template schema
+     * @param {Object} response
+     */
+    getTemplateSchemaInfo = (metadataQueryResponse: MetadataQueryResponse): Promise<MetadataTemplate | void> => {
+        this.metadataQueryResponse = this.filterMetdataQueryResponse(metadataQueryResponse);
+        const { entries } = metadataQueryResponse;
+
+        if (!entries || entries.length === 0) {
+            // if no items found for provided query, don't make API call to get template info
+            return Promise.resolve();
+        }
+
+        const metadata = getProp(entries, '[0].metadata');
```

Comment:
Can you add a constant for this string?

Reply:
This is object path `entries[0].metadata`.  Don't think we should create constant out of this as this is not going to be used elsewhere?